### PR TITLE
Make etcd-members-in-same-instance-group into a warning

### PR DESCRIPTION
**Pull request by justinsb**
This allows for simple single-AZ HA configurations.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1607)

### CODE REVIEW

File: upup/pkg/fi/cloudup/populate_cluster_spec.go [hunk 1]
```diff
@@ -152,8 +152,7 @@ func (c *populateClusterSpec) run() error {
 					instanceGroupName := fi.StringValue(m.InstanceGroup)
 
 					if etcdInstanceGroups[instanceGroupName] != nil {
-						// Maybe this should just be a warning
-						return fmt.Errorf("EtcdMembers are in the same InstanceGroup %q in etcd-cluster %q", instanceGroupName, etcd.Name)
+						glog.Warningf("EtcdMembers are in the same InstanceGroup %q in etcd-cluster %q", instanceGroupName, etcd.Name)
```
Comment (chrislovecnm) [inline upup/pkg/fi/cloudup/populate_cluster_spec.go]: Should we say why it is a warning?

**Comment by justinsb** [comment]
Updated message to add "(fault-tolerance may be reduced)"

**Comment by chrislovecnm** [comment]
e2e is still running

**Pull request merged**